admin/configuration.php

```diff
@@ -21,7 +21,17 @@
           $messageStack->add_session(ERROR_ADMIN_DEMO, 'caution');
           zen_redirect(zen_href_link(FILENAME_CONFIGURATION, 'gID=' . $_GET['gID'] . '&cID=' . (int)$cID));
         }
+        
         $configuration_value = zen_db_prepare_input($_POST['configuration_value']);
+        // See if there are any configuration checks 
+        $checks = $db->Execute("SELECT val_function FROM " . TABLE_CONFIGURATION . " WHERE configuration_id = '" . (int)$cID . "'");
+       if (!$checks->EOF) { 
+           require_once("includes/configuration_checks.php"); 
+           if (check_configuration($configuration_value, $checks->fields['val_function']) === false) {
+                $messageStack->add_session(TEXT_ADMIN_VALUE_OUT_OF_RANGE, 'caution');
+                zen_redirect(zen_href_link(FILENAME_CONFIGURATION, 'gID=' . $_GET['gID'] . '&cID=' . (int)$cID));
+              } 
+       }
```

Comment:
This is gonna sound nit-picky, but your indentation is all off, and doesn't match the section of code you're inserting this into.